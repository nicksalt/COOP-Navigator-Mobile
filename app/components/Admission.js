import React from "react";
import { View, Button, ScrollView, Text, TouchableOpacity } from "react-native";
import Toast from "react-native-easy-toast";
import CheckBox from "react-native-check-box";

export default class Admission extends React.Component {
  constructor() {
    super();
    this.state = {
      checked: false,
      studentAnswerText: "N/A",
      disabled: false
    };
  }
  updateTerms = () => {
    if (this.state.checked) {
      this.setState({
        checked: true,
        studentAnswerText: "Accepted",
        disabled: true
      });
    } else {
      this.refs.toast.show("Must accept Terms & Conditions");
    }
  };
  render() {
    return (
      <ScrollView>
        <Text style={styles.subtitle}>Enrollment Details</Text>
        <View style={styles.container}>
          <Text style={styles.field}>{"You're currently registered in:"}</Text>
          <Text style={styles.fieldValue}>
            Bachelor of Applied Science Software Engineering (Co-op)
          </Text>
          <Text style={styles.field}>Your current CO-OP status is:</Text>
          <Text style={styles.fieldValue}>Admitted</Text>
          <Text style={styles.field}>First expected work term:</Text>
          <Text style={styles.fieldValue}>2015, Summer</Text>
          <Text style={styles.field}>Special condition of admission:</Text>
          <Text style={styles.fieldValue}>
            Must maintain a minimum CGPA of 4.5. Must complete all courses
            currently registered 2014-2015. 14/10/07 TN
          </Text>
          <Text style={styles.fieldValueBlock}>
            Please note that this admission offer is CONDITIONAL.
          </Text>
          <Text style={styles.fieldValueBlock}>
            To Keep your spot in CO-OP:
          </Text>
          <Text style={styles.fieldValue}>
            1) You Need to attend the first mandatory workshop before accepting
            your offer of admission. If you miss the first workshop, your offer
            of admission may be revoked. {"\n"}
            {"\n"}
            2) Your CGPA must be above the minimum required CGPA for your
            program following the current study semester.
          </Text>
        </View>
        <Text style={styles.subtitle}>Confirming Participation</Text>
        <View style={styles.container}>
          <Text style={[styles.field, styles.studentAnswer]}>
            {this.state.studentAnswerText}
          </Text>
          <View style={styles.checkBoxLayout}>
            <CheckBox
              isChecked={this.state.checked}
              onClick={() => this.setState({ checked: !this.state.checked })}
              disabled={this.state.disabled}
            />
            <Text style={styles.checkBoxfieldValue}>
              *I have read the CO-OP Admission Agreement and agree to comply
              with the terms and conditions.
            </Text>
          </View>
          <Toast ref="toast" />
          <TouchableOpacity
            style={[
              !this.state.disabled && styles.buttonContainer,
              this.state.disabled && styles.buttonContainerDisabled
            ]}
            onPress={this.updateTerms}
            title="Accept"
            disabled={this.state.disabled}
          >
            <Text style={styles.acceptButton}>ACCEPT</Text>
          </TouchableOpacity>
        </View>
      </ScrollView>
    );
  }
}

const styles = {
  container: {
    padding: 10
  },
  subtitle: {
    padding: 10,
    fontSize: 16,
    fontWeight: "bold",
    textAlign: "center",
    color: "white",
    backgroundColor: "#a9343a"
  },
  field: {
    fontSize: 16,
    fontWeight: "bold"
  },
  fieldValue: {
    fontSize: 15,
    paddingBottom: 10
  },
  fieldValueBlock: {
    fontSize: 15,
    fontWeight: "bold",
    paddingBottom: 10
  },
  studentAnswer: {
    textAlign: "center"
  },
  checkBoxLayout: {
    flexDirection: "row"
  },
  checkBoxfieldValue: {
    flex: 1,
    flexWrap: "wrap"
  },
  buttonContainer: {
    margin: 50,
    backgroundColor: "#d8d8d8",
    borderRadius: 15,
    padding: 15
  },
  buttonContainerDisabled: {
    margin: 50,
    backgroundColor: "hsl(0,25%,90%)",
    borderRadius: 15,
    padding: 15
  },
  acceptButton: {
    textAlign: "center",
    fontWeight: "bold",
    color: "#a9343a"
  }
};
