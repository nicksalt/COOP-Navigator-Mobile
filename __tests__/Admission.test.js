import React from "react";
import Admission from "../app/components/Admission";
import renderer from "react-test-renderer";

jest.mock("ScrollView", () => "ScrollView");

test("Admission snapshot renders correctly", () => {
  const tree = renderer.create(<Admission />).toJSON();
  expect(tree).toMatchSnapshot();
});
