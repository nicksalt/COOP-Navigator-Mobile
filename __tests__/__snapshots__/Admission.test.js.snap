// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Admission snapshot renders correctly 1`] = `
<ScrollView>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "#a9343a",
        "color": "white",
        "fontSize": 16,
        "fontWeight": "bold",
        "padding": 10,
        "textAlign": "center",
      }
    }
  >
    Enrollment Details
  </Text>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      You're currently registered in:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
          "paddingBottom": 10,
        }
      }
    >
      Bachelor of Applied Science Software Engineering (Co-op)
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Your current CO-OP status is:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
          "paddingBottom": 10,
        }
      }
    >
      Admitted
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      First expected work term:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
          "paddingBottom": 10,
        }
      }
    >
      2015, Summer
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Special condition of admission:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
          "paddingBottom": 10,
        }
      }
    >
      Must maintain a minimum CGPA of 4.5. Must complete all courses currently registered 2014-2015. 14/10/07 TN
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
          "fontWeight": "bold",
          "paddingBottom": 10,
        }
      }
    >
      Please note that this admission offer is CONDITIONAL.
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
          "fontWeight": "bold",
          "paddingBottom": 10,
        }
      }
    >
      To Keep your spot in CO-OP:
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 15,
          "paddingBottom": 10,
        }
      }
    >
      1) You Need to attend the first mandatory workshop before accepting your offer of admission. If you miss the first workshop, your offer of admission may be revoked. 
      

      

      2) Your CGPA must be above the minimum required CGPA for your program following the current study semester.
    </Text>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "#a9343a",
        "color": "white",
        "fontSize": 16,
        "fontWeight": "bold",
        "padding": 10,
        "textAlign": "center",
      }
    }
  >
    Confirming Participation
  </Text>
  <View
    style={
      Object {
        "padding": 10,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
          },
          Object {
            "textAlign": "center",
          },
        ]
      }
    >
      N/A
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        isTVSelectable={true}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={null}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <Image
            source={1}
            style={
              Object {
                "tintColor": undefined,
              }
            }
          />
        </View>
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "flex": 1,
            "flexWrap": "wrap",
          }
        }
      >
        *I have read the CO-OP Admission Agreement and agree to comply with the terms and conditions.
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "#d8d8d8",
          "borderRadius": 15,
          "margin": 50,
          "opacity": 1,
          "padding": 15,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#a9343a",
            "fontWeight": "bold",
            "textAlign": "center",
          }
        }
      >
        ACCEPT
      </Text>
    </View>
  </View>
</ScrollView>
`;
